src/SignalRServiceExtension/Bindings/SignalRInputBindings/AttributeCloner.cs

```diff
@@ -0,0 +1,532 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+using System.Reflection;
+using Microsoft.Azure.WebJobs.Description;
+using Microsoft.Azure.WebJobs.Host;
+using Microsoft.Azure.WebJobs.Host.Bindings;
+using Microsoft.Azure.WebJobs.Host.Bindings.Path;
+using Microsoft.Extensions.Configuration;
+using Newtonsoft.Json;
+using BindingData = System.Collections.Generic.IReadOnlyDictionary<string, object>;
+using BindingDataContract = System.Collections.Generic.IReadOnlyDictionary<string, System.Type>;
+// Func to transform Attribute,BindingData into value for cloned attribute property/constructor arg
+// Attribute is the new cloned attribute - null if constructor arg (new cloned attr not created yet)
+using BindingDataResolver = System.Func<System.Attribute, System.Collections.Generic.IReadOnlyDictionary<string, object>, object>;
+
+using Validator = System.Action<object>;
+
+#pragma warning disable CS0618 // Type or member is obsolete
+namespace Microsoft.Azure.WebJobs.Extensions.SignalRService
+{
+    // Clone an attribute and resolve it.
+    // This can be tricky since some read-only properties are set via the constructor.
+    // This assumes that the property name matches the constructor argument name.
+    internal class AttributeCloner<TAttribute>
```

Comment:
>internal class AttributeCloner<TAttribute> [](start = 4, length = 42)

Add comments about the file is copied from official external, so if further issue found we'd easier look for source. [https://github.com/Azure/azure-webjobs-sdk/blob/dev/src/Microsoft.Azure.WebJobs.Host/Bindings/AttributeCloner.cs](https://github.com/Azure/azure-webjobs-sdk/blob/dev/src/Microsoft.Azure.WebJobs.Host/Bindings/AttributeCloner.cs)